sqld/src/database/write_proxy/replication.rs

```diff
@@ -305,4 +308,29 @@ impl ReadReplicationHook {
 
         Ok(())
     }
+
+    #[allow(dead_code)]
+    fn is_valid_new_gen(&mut self, gen: u64, gen_start_frame_index: u64) -> bool {
+        if self.current_gen != gen {
+            if let Some(last_applied) = self.last_applied_index {
+                if last_applied <= gen_start_frame_index {
+                    // TODO persist to file.
```

Comment:
I think for the first iteration it's totally fine to just keep current_gen in memory. Replicas don't use persistent disks so if the machine is restarted we probably loose this info anyway.

Reply:
That's true for turso, but sqld should target more general deployment scenario, given that it's an open source project

Reply:
Good point. Still probably not absolutely required in the first iteration.